driver-core/src/test/unit/com/mongodb/client/model/AggregatesSpecification.groovy

```diff
@@ -69,6 +70,28 @@ import static org.bson.BsonDocument.parse
 
 class AggregatesSpecification extends Specification {
 
+    @IgnoreIf({ !serverVersionAtLeast(4, 3) })
+    def 'should render $accumulator'() {
+        given:
+        def initFunction = 'function() { return { count : 0, sum : 0 } }';
+        def accumulateFunction = 'function(state, numCopies) { return { count : state.count + 1, sum : state.sum + numCopies } }';
+        def mergeFunction = 'function(state1, state2) { return { count : state1.count + state2.count, sum : state1.sum + state2.sum } }';
+        def finalizeFunction = 'function(state) { return (state.sum / state.count) }';
+
+        expect:
+        toBson(accumulator(initFunction, accumulateFunction, mergeFunction)) ==
```

Comment:
It's tedious, but let's have tests for all the overloads, with null and non-null values for all the nullable parameters

Reply:
Done